sdk/eventhub/azure-eventhubs/azure/eventhub/aio/_consumer_client_async.py

```diff
@@ -0,0 +1,187 @@
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+import logging
+from typing import Any, Union, TYPE_CHECKING, Callable, Dict, List, Tuple
+from azure.eventhub import EventPosition, EventData, \
+    EventHubSharedKeyCredential, EventHubSASTokenCredential
+from .eventprocessor.event_processor import EventProcessor, CloseReason
+from .eventprocessor.partition_context import PartitionContext
+from .client_async import EventHubClient
+if TYPE_CHECKING:
+    from azure.core.credentials import TokenCredential  # type: ignore
+
+log = logging.getLogger(__name__)
+
+
+class EventHubConsumerClient(EventHubClient):
+    """Represents an AMQP connection to an EventHub and receives event data from it.
+
+    The main goal of `EventHubConsumerClient` is to receive events from all partitions of an EventHub with
+    load balancing and checkpointing.
+
+    When multiple `EventHubConsumerClient` works with one process, multiple processes, or multiple computer machines
+    and if they use the same repository as the load balancing and checkpointing store, they will balance automatically.
+    To enable the load balancing and / or checkpointing, partition_manager must be set when creating the
+    `EventHubConsumerClient`.
+
+    An `EventHubConsumerClient` can also receive from a specific partition when you call its method `receive()`
+    and specify the partition_id.
+    Load balancing won't work in single-partition mode. But users can still save checkpoint if the partition_manager
+    is set.
+
+    Example:
+        .. literalinclude:: ../samples/test_examples_eventhub_async.py
+            :start-after: [START create_eventhub_consumer_client_async]
+            :end-before: [END create_eventhub_consumer_client_async]
+            :language: python
+            :dedent: 4
+            :caption: Create a new instance of the EventHubConsumerClient.
+    """
+
+    def __init__(self, host, event_hub_path, credential, **kwargs):
+        # type:(str, str, Union[EventHubSharedKeyCredential, EventHubSASTokenCredential, TokenCredential], Any) -> None
+        """
+        :param host: The hostname of the Event Hub.
+        :type host: str
+        :param event_hub_path: The path of the specific Event Hub to connect the client to.
+        :type event_hub_path: str
+        :param credential: The credential object used for authentication which implements particular interface
+         of getting tokens. It accepts ~azure.eventhub.EventHubSharedKeyCredential,
+         ~azure.eventhub.EventHubSASTokenCredential, credential objects generated by the azure-identity library and
+         objects that implement get_token(self, *scopes) method.
+        :keyword network_tracing: Whether to output network trace logs to the logger. Default
+         is `False`.
+        : network_tracing: bool
```

Comment:
keyword I guess?

Reply:
Oui!

Reply:
Updated async docstring together with this one